src/seedsigner/views/view.py

```diff
@@ -243,3 +243,50 @@ def run(self):
         ).display()
         
         return Destination(MainMenuView, clear_history=True)
+
+class MicroSDToastView(View):
+    def event_handler(self, action, mount_dir):
+        import time
+        
+        self.current_screen = self.renderer.canvas.copy()
+        
+        # Special case when screensaver is running
+        if self.controller.screensaver._is_running:
+            self.buttons.override_ind = True
+        
+        # MicroSDToastView blocks any attempts to use the Renderer in another thread so it
+        # never gives up the lock until it returns.
+        with self.renderer.lock:
+            if action == "remove":
+
+                toast = ToastOverlay(
+                    icon_name=FontAwesomeIconConstants.SDCARD,
+                    color=GUIConstants.NOTIFICATION_COLOR,
+                    label_text="MicroSD removed"
+                )
+                toast.render()
+                self.renderer.show_image()
+                
+                t_end = time.time() + 3
+                while time.time() < t_end:
+                    if self.buttons.has_any_input():
+                        break
+
+                self.renderer.show_image(self.current_screen)
+            
+            elif action == "add":
+                
+                toast = ToastOverlay(
+                    icon_name=FontAwesomeIconConstants.SDCARD,
+                    color=GUIConstants.NOTIFICATION_COLOR,
+                    label_text="MicroSD inserted"
+                )
+                toast.render()
+                self.renderer.show_image()
+                
+                t_end = time.time() + 3
+                while time.time() < t_end:
+                    if self.buttons.has_any_input():
+                        break
```

Comment:
I think this hold should also be moved into the `ToastOverlay` to give it explicit control of the UI and any further input until it returns, just like a normal `Screen` behaves.